src/sentry_core.c

```diff
@@ -213,6 +213,22 @@ sentry_clear_modulecache(void)
     sentry__modulefinder_cleanup();
 }
 
+int
+sentry_reset_backend(void)
+{
+    sentry_backend_t *backend = g_options->backend;
```

Comment:
You have to wrap this in `SENTRY_WITH_OPTIONS (options) {`
Otherwise you are not thread safe, and another thread can just free `g_options` from under you.
Also note that you can’t early-return from that macro :-(